frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1887,6 +1891,93 @@
             });
         };
 
+        vm.togglePhaseSplitVisibility = function (ev) {
+            ev.stopPropagation();
+            var togglePhaseSplitState;
+            if(vm.phaseSplitVisibility.state == "Public")
+                togglePhaseSplitState = "private";
+            else
+                togglePhaseSplitState = "public";
+
+            var confirmDialog = $mdDialog.confirm()
+                          .title('Make the phase split state ' + togglePhaseSplitState + '?')
+                          .ariaLabel('')
+                          .targetEvent(ev)
+                          .ok('Yes, I\'m sure')
+                          .cancel('No');
+
+            $mdDialog.show(confirmDialog).then(function () {
+                parameters.url = "challenges/challenge/create/challenge_phase_split/" + vm.phaseSplitId + "/";
+                parameters.method = 'PATCH';
+                if (vm.phaseSplitVisibility.state == "Public") {
+                    parameters.data = {
+                        "visibility": 1,
+                    };
+                } else {
+                    parameters.data = {
+                        "visibility": 3,
+                    };
+                }
+
+                parameters.callback = {
+                    onSuccess: function(response) {
+                        var status = response.status;
+                        if (status === 200) {
+                            if(vm.phaseSplitVisibility.state == "Public"){
+                                vm.phaseSplitVisibility.state = "Private";
+                                vm.phaseSplitVisibility.icon = "fa fa-eye-slash red-text";
```

Comment:
`fa` need not be in branching logic, let it be in the html tag itself.

Reply:
@galipremsagar ah, I thought of changing the whole class name instead of leaving just `fa` there because there are some font-awesome icons which starts with like `fas` or `fal` also, so I thought it would be easy just to replace the whole class name at one place only. Please let me know what do you think about it